src/tokens.rs

```diff
@@ -456,15 +484,77 @@ pub fn test_process_distribute_with_client<C: Client>(client: C, sender_keypair:
         sender_keypair: Some(Box::new(sender_keypair)),
         fee_payer: Some(Box::new(fee_payer)),
         dry_run: false,
-        bids_csv,
+        allocations_csv: false,
+        input_csv,
+        transactions_db: transactions_db.clone(),
+        dollars_per_sol: Some(0.22),
+    };
+    process_distribute_tokens(&thin_client, &args).unwrap();
```

Comment:
This test is a lot of copy pasta for essentially a subset of the same functionality. Can you move the bids/allocations_csv into a small fiction `read_allocations(args)` so that you just need to test that either branch produces a `Vec<Allocation>`?